smarts/core/sumo_traffic_simulation.py

```diff
@@ -295,14 +295,24 @@ def setup(self, next_scenario) -> ProviderState:
         return self._compute_provider_state()
 
     def _close_traci_and_pipes(self):
+        """We should expect this method to always work without throwing"""
+
+        def __safe_close(conn):
+            try:
+                conn.close()
+            except:
+                pass
+
         if self._sumo_proc:
-            self._sumo_proc.stdin.close()
-            self._sumo_proc.stdout.close()
-            self._sumo_proc.stderr.close()
+            __safe_close(self._sumo_proc.stdin)
+            __safe_close(self._sumo_proc.stdout)
+            __safe_close(self._sumo_proc.stderr)
```

Comment:
why is this being tried 3 times?

Reply:
There are 3 different streams here that need to be closed: `stdin`, `stdout`, `stderr`.